features/FEATURE_BLE/ble/gap/Gap.h

```diff
@@ -538,6 +538,18 @@ class Gap {
      */
     virtual uint16_t getMaxAdvertisingDataLength();
 
+    /** Return maximum advertising data length supported for connectable advertising.
+     *
+     * @return Maximum advertising data length supported for connectable advertising.
+     */
+    virtual uint16_t getMaxConnectableAdvertisingDataLength();
+
+    /** Return maximum advertising data length you may set if advertising set is active.
+     *
+     * @return Maximum advertising data length you may set if advertising set is active.
+     */
+    virtual uint8_t getMaxActiveSetAdvertisingDataLength();
```

Comment:
I would use a `uint16_t` as the return type here. The Bluetooth SIG may lift that limitation in a next release.

Reply:
the update needs to happen in a single HCI command which is limited to 255 bytes

Reply:
It is understood that for now the update must happen in a single hci packet when advertising is on. 
However that may change in the future as the controller can easily identify when the update is ready to be commited when the host sends a _last fragment_ operation. 

That's why I suggest using `uint16_t` as the return type. It virtually cost nothing and it can avoid integer size issues if the Bluetooth specification is updated.

Reply:
On ARM, any narrow types are normally more expensive than 32-bit types, if they're automatic variables, arguments or return values. The compiler needs to add extra narrowing instructions.

Unless the narrow width is semantically important, it's best to stick with `int`, `int32_t` or `size_t` most of the time. (I used to avoid `size_t` when still dealing with 8-bit CPUs where that forced 32-bit rather than 16-bit arithmetic, but no longer.)  

IIRC, standalone static variables are also normally word-aligned anyway, so I don't think narrow types buy you anything there either. Automatic variables are are also word-aligned on the stack.

So narrow types are only really useful in structures and arrays, unless you require narrow unsigned modulo arithmetic.